components/brave_welcome_ui/components/screens/searchBox.tsx

```diff
@@ -18,11 +19,64 @@ interface Props {
   index: number
   currentScreen: number
   onClick: () => void
+  onChange: (event: React.ChangeEvent<HTMLSelectElement>) => void
+  isDefaultSearchGoogle: boolean
 }
 
-export default class SearchEngineBox extends React.PureComponent<Props, {}> {
+interface State {
+  searchEngineSelected: boolean,
+  searchEngineIndex: number,
+  searchProviders: Array<any>
+}
+
+interface SearchEngineEntry {
```

Comment:
Example entry (returned from the `SearchEnginesHandler`):
```json
{
  "canBeDefault": true,
  "canBeEdited": true,
  "canBeRemoved": true,
  "default": false,
  "displayName": "DuckDuckGo",
  "iconURL": "https://duckduckgo.com/favicon.ico",
  "id": 3,
  "isOmniboxExtension": false,
  "keyword": ":d",
  "modelIndex": 1,
  "name": "DuckDuckGo",
  "url": "https://duckduckgo.com/?q=%s&t=brave",
  "urlLocked": true
}
```

Reply:
@imptrx could you move this interface to welcome.d.ts file?